### PR TITLE
Add some elaboration where `...` operator appear according to C++11.

### PR DESCRIPTION
**Pull request by p1v0t**

### CODE REVIEW

**Comment by AnthonyCalandra** [comment]
I think the example code snippets cover these cases... no?

**Comment by p1v0t** [comment]
Exactly, I just tried to add what's just done on the example. Like `sizeof...` operator which is not mentioned. In fact in C++11, the one need to make recursive call to use variadics which C++17 relaxed it with unary and binary variadic calls.

**Comment by AnthonyCalandra** [comment]
In my opinion, I don't think the first three bullet points provide much value. The explanations themselves are pretty basic and I think the code snippets really describe what they're doing succinctly.

I think the sizeof bullet point would be great as an inline comment. since I can see some confusion between how sizeof works on types that are not packs.

Also, be sure to change both CPP11.md and README.md.

**Comment by p1v0t** [comment]
Okay, got it... Thank you very much :smiley_cat:

File: CPP11.md [hunk 1]
```diff
@@ -120,11 +120,11 @@ f(std::move(z)); // deduces as f(int&& &&) => f(int&&)
 See also: [`std::move`](#stdmove), [`std::forward`](#stdforward), [`rvalue references`](#rvalue-references).
 
 ### Variadic templates
-The `...` syntax creates a _parameter pack_ or expands one. A template _parameter pack_ is a template parameter that accepts zero or more template arguments (non-types, types, or templates). A template with at least one parameter pack is called a _variadic template_.
+The `...` syntax creates a _parameter pack_ or expands one. A template _parameter pack_ is a template parameter that accepts zero or more template arguments (non-types, types, or templates). A template with at least one parameter pack is called a _variadic template_. The parameter pack must be at the end of argument list.
```
Comment (AnthonyCalandra) [inline CPP11.md]: > The parameter pack must be at the end of **the** argument list.

File: CPP11.md [hunk 2]
```diff
@@ -120,11 +120,11 @@ f(std::move(z)); // deduces as f(int&& &&) => f(int&&)
...
 ```c++
 template <typename... T>
 struct arity {
-  constexpr static int value = sizeof...(T);
+  constexpr static int value = sizeof...(T); // sizeof... gives num. or args in parameter pack
```
Comment (AnthonyCalandra) [inline CPP11.md]: Move the comment to its own line:
```
// `sizeof...` returns the number of args in the parameter pack
```

File: CPP11.md (hunk 1 above)
Comment (AnthonyCalandra) [reply CPP11.md]: This typo still remains ^

File: CPP11.md [hunk 3]
```diff
@@ -120,11 +120,11 @@ f(std::move(z)); // deduces as f(int&& &&) => f(int&&)
 See also: [`std::move`](#stdmove), [`std::forward`](#stdforward), [`rvalue references`](#rvalue-references).
 
 ### Variadic templates
-The `...` syntax creates a _parameter pack_ or expands one. A template _parameter pack_ is a template parameter that accepts zero or more template arguments (non-types, types, or templates). A template with at least one parameter pack is called a _variadic template_.
+The `...` syntax creates a _parameter pack_ or expands one. A template _parameter pack_ is a template parameter that accepts zero or more template arguments (non-types, types, or templates). A template with at least one parameter pack is called a _variadic template_. The parameter pack must be at the end of argument list.
 ```c++
 template <typename... T>
 struct arity {
-  constexpr static int value = sizeof...(T);
+  constexpr static int value = sizeof...(T); // `sizeof...` returns the number of arguments in the parameter pack
```
Comment (AnthonyCalandra) [inline CPP11.md]: Forgot to mention in my latest review to move the comment to its own line:
```
// `sizeof...` returns the number of arguments in the parameter pack
constexpr static int value = sizeof...(T);
```

**Comment by AnthonyCalandra** [comment]
?

**Pull request closed without merging**